### PR TITLE
SickChill use a pypi based install - from 5431

### PR DESCRIPTION
**Pull request by BKSteve**
## Description

SickChill update test with #5431 

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Bug fix
- [ ] New Package
- [x] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

### Tested
- [x] x64-6.1
- [x] x64-7.0
- [x] armv7-6.1
- [x] armv7-7.0
- [x] evansport-6.1
- [x] evansport-7.0

### CODE REVIEW

**Comment by th0ma7** [comment]
Have you got it working?  If so, what was missing (seing the last build being succesful)

**Comment by BKSteve** [comment]
Waiting for others not using x64 to report back. 
Everything else pretty clear with the changes

**Comment by BKSteve** [comment]
@th0ma7 ultimately it was getting the requirements sorted out. (a couple of typos and bits)
Need to cross/cryptography (copied from your python 310 makefile)
Then a sickchill pyproject.toml clean up.

As of the last build using 20221013 and writing this have 1 positive working on a DS114 armv7 DSM6 
Everybody with x64 is of course working. I've tested on both DSM6 and DSM7 x64 devices.

**Comment by BKSteve** [comment]
The `pip-cache` going into /var or /\@appdata is noted. 

Wondered why `cachedir=${SYNOPKG_PKGVAR}/pip-cache` in `spksrc.service.installer.functions` is `SYNOPKG_PKGVAR` rather than say `SYNOPKG_PKGDEST/share`

As this puts the cached pip items in the user variable location rather than package location. (\@appdata not \@appstore)

**Comment by BKSteve** [comment]
We have x64, armv7 and evansport confirmed working on both DSM 6 & 7.

@th0ma7 grateful if you could run you eyes over this and then merge to master.

File: spk/sickchill/src/service-setup.sh [hunk 1]
```diff
@@ -62,10 +55,12 @@ service_postupgrade() {
 
 service_preupgrade ()
 {
-    # We have to reset /env folder to 3.8 so remove entire folder as it gets rebuilt in postinst and this avoids any conflicts.
-    # Revision 1 was python 3.7. For cleaner update remove share and lib folders for clean install, leave user data /var.
+    # We have to reset /env for Python and package changes, it gets rebuilt in postinst and this avoids any conflicts.
+    # For cleaner update remove bin, env, share and lib folders for fresh install, leave user data /var & /@appdata
```
Comment (th0ma7) [inline spk/sickchill/src/service-setup.sh:59]: I'm surprised by this as normally the entire `target/*` directory (e.g. `${SYNOPKG_PKGDEST}) gets deleted at upgrade time.  Does this really have an effect?

File: spk/sickchill/src/requirements-crossenv.txt [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-greenlet==1.1.2
-lxml==4.6.4
-MarkupSafe==2.0.1
-msgpack==1.0.2
-PyNaCl==1.4.0
-SQLAlchemy==1.4.26
-tornado==6.1
-wrapt==1.13.3
+cffi==1.15.1
+cryptography==38.0.1
```
Comment (th0ma7) [inline spk/sickchill/src/requirements-crossenv.txt:2]: Now I noticed this, so you're building both `py310-py310` and a `py36-abi3` wheels.  Therefore you end-up having two versions of the same wheel in the `work-<arch>-<ver>/wheelhouse/cryptography*.whl`.  Only one is needed.

Can you check which version was used on your tested installations as you should only be keeping one or the other.

**Review by th0ma7 — CHANGES_REQUESTED** [review]
Overall LGTM besides building duplicate `cryptography` wheels (see comments below).

**Comment by th0ma7** [comment]
> The `pip-cache` going into /var or /@appdata is noted.
> 
> Wondered why `cachedir=${SYNOPKG_PKGVAR}/pip-cache` in `spksrc.service.installer.functions` is `SYNOPKG_PKGVAR` rather than say `SYNOPKG_PKGDEST/share`
> 
> As this puts the cached pip items in the user variable location rather than package location. (@appdata not @appstore)

Previously everything was under the `target` directory (e.g. `SYNOPKG_PKGDEST`).  Starting with DSM7 you can uninstall a package without uninstalling the user-files and upgrade also now avoid impacting user-files as well (on DSM6 we needed to temporarily copy them elsewhere).

For all theses reasons, having the `pip-cache` located there makes it relevant as otherwise it would get deleted at every package upgrade, thus removing the interest of caching in the first place.  By being under the `var` directory, on older much slower targets, it fasten the upgrade process significantly.

**Comment by th0ma7** [comment]
Sorry for the very limited cycles this week (and on my cellphone).  Try setting up `SERVICE_USER = AUTO` in `Makefile`.

**Comment by BKSteve** [comment]
Will try with `auto` changed to `AUTO` in makefile as already there with lowercase.

Is there any way to modify the `service-setup.sh` file and build a new package without going through the full build process with `packages` ?
So after the build and everything in `work` folder just tweak service-setup.sh and build new SPK.

**Comment by th0ma7** [comment]
Awesome, great work!  And yes you can simply use `make spkclean` or `wheelclean` to invoke rebuilds.

Leave me a few day before doing a review + merge + publishing.  Ping me by EOW if you don't hear back from me.

**Review by th0ma7 — APPROVED** [review]

**Comment by th0ma7** [comment]
awesome work, thnx for your contribution!
now merged and building + publishing in progress

File: spk/sickchill/Makefile [hunk 3]
```diff
@@ -1,32 +1,36 @@
...
 STARTABLE = yes
 DISPLAY_NAME = SickChill
-CHANGELOG = "1. Migration to Python 3.10<br/>2. SickChill environment fix<br/>3. Added DSM7 Support<br/>4. cryptography resolution for armv7"
+CHANGELOG = "Switched to a wheel based install, as source and git installs are no longer supported by upstream."
 
 HOMEPAGE   = https://sickchill.github.io/
 LICENSE    = GPLv3+
 
-SERVICE_USER = auto
+SERVICE_USER = AUTO
```
[the remark this answers is not in the document]
Comment (th0ma7) [reply spk/sickchill/Makefile:33]: I'm now not so sure.  @hgy59 may know more about it.  In the meantime tomorrow i'll disable the packages on-line until we figure it out.

**Comment by BKSteve** [comment]
Sorry think I messed this up. #5458 
Did this so I can pull the built packages zip.

File: spk/sickchill/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (hgy59) [reply spk/sickchill/Makefile:33]: IMHO `auto` is the only valid value for `SERVICE_USER`.

The install log entries (with `SERVICE_USER = auto`) are correct showing `... USER=sickchill GROUP=sc-download SHARE_PATH=` 
`$USER` is the base name and as variable for the user any install script must use `$EFF_USER` (is either `sc-sickchill` for DSM >= 6 or `svc-sickchill` for DSM < 6.

[the remark this answers is not in the document]
Comment (miigotu) [reply spk/sickchill/Makefile:33]: As I suspected. The question is, how to correct ownership when keeping data between upgrades from dsm6=>dsm7?

Also, is there a proper location to store our database and configuration, or at least backups, outside the spk install directory?

Thanks.

**Pull request merged**